lib/elixir/lib/calendar/duration.ex

```diff
@@ -299,6 +299,33 @@ defmodule Duration do
     }
   end
 
+  @doc """
+  Compares two durations.
+
+  Returns `0` if the durations are equal, `-1` if the first duration is less than the second, and `1` if the first duration is greater than the second.
+
+  ## Examples
+
+      iex> Duration.compare(Duration.new!(second: 10), Duration.new!(second: 20))
+      -1
+      iex> Duration.compare(Duration.new!(second: 20), Duration.new!(second: 10))
+      1
+      iex> Duration.compare(Duration.new!(second: 10), Duration.new!(second: 10))
+      0
+  """
+  @doc since: "1.19.0"
+  @spec compare(t, t) :: integer
+  def compare(%Duration{} = d1, %Duration{} = d2) do
+    d1_timeout = Kernel.to_timeout(d1)
+    d2_timeout = Kernel.to_timeout(d2)
```

Comment:
This is not considering microseconds

Reply:
oops

Reply:
I think I did it correctly ... but I am unsure how to deal with leap years; any math suggestion here 😭